woocommerce/woocommerce-gutenberg-products-block
# Unify Chip styles

Fixes #1786.

This PR updates the Filter Products by Attribute and Active Filters blocks so they use the `Chip` component introduced in the Cart & Checkout blocks. `Chip` needed some changes so it could be used in those scenarios.

I also took the chance to fix some theme compatibility issues in those blocks (input backgrounds & Clear All color).

We are making some style changes and switching some components, so there is the possibility of themes breakage but I think it's quite small. IMO there is no need for a dev note. Thoughts?

#### Accessibility


### Screenshots
Screenshots with Bistro theme:

_Before:_
![imatge](https://user-images.githubusercontent.com/3616980/85393809-e65a2c80-b54d-11ea-801e-2685484459ca.png)

_After:_
![imatge](https://user-images.githubusercontent.com/3616980/85393664-aabf6280-b54d-11ea-933e-5ca448c45aba.png)

### How to test the changes in this Pull Request:
0. Ideally, set your theme background to something different from white so you can verify inputs are still legible.
1. Create a page with the All Products block and the Filter Products by Attribute (set the attributes to Query Type: OR, Display Style: Dropdown) and Active Filters (Display Style: Chips).
2. Filter Products by Attribute:
  2.1. Try adding new filters.
  2.2. Verify chips have the correct styles.
  2.3. Try removing them with the keyboard (backspace or <kbd>Del</kbd>).
  2.4. Try removing them clicking on the chip name.
3. Active Filters:
  3.1 Verify chips have the correct styles.
4. Catching regressions:
  4.1. Verify there are no regression in the Filter Products by Attribute and the Active Filters blocks with other attribute combinations: verify everything is still working and there are no visual bugs.
  4.2. Verify there are no regressions with the Chips in the Cart: try adding a coupon in the Cart or Checkout blocks and verify it still looks correct.

### Changelog

> Chip styles of the Filter Products by Attribute and Active Filters have been updated to give a more consistent experience.
